browser/ui/webui/brave_welcome_ui.cc

```diff
@@ -67,8 +68,15 @@ BraveWelcomeUI::BraveWelcomeUI(content::WebUI* web_ui, const std::string& name)
   auto handler_owner = std::make_unique<WelcomeDOMHandler>();
   WelcomeDOMHandler* handler = handler_owner.get();
   web_ui->AddMessageHandler(std::move(handler_owner));
-  handler->Init();
+
   Profile* profile = Profile::FromWebUI(web_ui);
+
+  // added to allow front end to read/modify default search engine
+  auto search_handler_owner = std::make_unique<settings::SearchEnginesHandler>(profile);
+  web_ui->AddMessageHandler(std::move(search_handler_owner));
+
+  handler->Init();
```

Comment:
Just a nit:  `Init()` and `OnWalletInitialized()` of `WelcomeDOMHandler` can be deleted.
`Init()` is empty method and the other doesn't have definition.

Reply:
Done! 👍